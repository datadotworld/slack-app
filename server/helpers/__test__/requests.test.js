const requestsHelpers = require('../requests')
const fixtures = require('../../jest/fixtures')

afterEach(() => {
  jest.clearAllMocks()
})

describe('requests helper methods', () => {
  describe('getAuthorizationRequestSlackBlocks', () => {
    it('should return correctly formatted blocks for authorizations created', () => {
      const result = requestsHelpers.getAuthorizationRequestSlackBlocks(
        fixtures.authorizationRequestCreatedEventBody
      )
      expect(result).toMatchSnapshot() 
    })

    it('should return correctly formatted blocks for authorizations cancelled/rejected', () => {
      const result = requestsHelpers.getAuthorizationRequestSlackBlocks(
        fixtures.authorizationRequestCancelledEventBody
      )
      expect(result).toMatchSnapshot() 
    })
  })

  describe('getContributionRequestSlackBlocks', () => {
    it('should return correctly formatted blocks for contributions created', () => {
      const result = requestsHelpers.getContributionRequestSlackBlocks(
        fixtures.contributionRequestCreatedEventBody
      )
      expect(result).toMatchSnapshot() 
    })

    it('should return correctly formatted blocks for contributions cancelled/rejected', () => {
      const result = requestsHelpers.getContributionRequestSlackBlocks(
        fixtures.contributionRequestCancelledEventBody
      )
      expect(result).toMatchSnapshot() 
    })
  })
})
