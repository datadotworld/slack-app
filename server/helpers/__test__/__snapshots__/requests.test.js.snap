// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`requests helper methods getAuthorizationRequestSlackBlocks should return correctly formatted blocks for authorization invites created 1`] = `
Array [
  Object {
    "elements": Array [
      Object {
        "text": "üìç Use Case Org",
        "type": "mrkdwn",
      },
    ],
    "type": "context",
  },
  Object {
    "text": Object {
      "text": "_<http://localhost:3000/patrick|Patrick Star>_ has been invited to have _edit_ access to _<http://localhost:3000/use-case-org/use-case-discvoerable|use-case-discvoerable>_ by _<http://localhost:3000/user8888|testing!>_",
      "type": "mrkdwn",
    },
    "type": "section",
  },
  Object {
    "elements": Array [
      Object {
        "action_id": "authorization_request.cancel",
        "style": "danger",
        "text": Object {
          "text": "Cancel",
          "type": "plain_text",
        },
        "type": "button",
        "value": "{\\"requestid\\":\\"5dee1394-5473-4329-aae2-f7bd46f5bfad\\",\\"agentid\\":\\"use-case-org\\",\\"datasetid\\":\\"use-case-discvoerable\\"}",
      },
    ],
    "type": "actions",
  },
  Object {
    "type": "divider",
  },
  Object {
    "fields": Array [
      Object {
        "text": "*<http://localhost:3000/patrick|Patrick Star>*
*User ID*: _patrick_
*E-mail*: _patrick@star.com_
",
        "type": "mrkdwn",
      },
      Object {
        "text": "*<http://localhost:3000/use-case-org/use-case-discvoerable|use-case-discvoerable>*
*Resource ID*: _use-case-discvoerable_
*Owner ID*: _use-case-org_
",
        "type": "mrkdwn",
      },
    ],
    "type": "section",
  },
  Object {
    "type": "divider",
  },
]
`;

exports[`requests helper methods getAuthorizationRequestSlackBlocks should return correctly formatted blocks for authorizations cancelled/rejected 1`] = `
Array [
  Object {
    "elements": Array [
      Object {
        "text": "üìç Use Case Org",
        "type": "mrkdwn",
      },
    ],
    "type": "context",
  },
  Object {
    "text": Object {
      "text": "_<https://ddw-corewebapp.dev.data.world/patrick|Patrick Star>_ has cancelled their request for _view_ access to _<https://ddw-corewebapp.dev.data.world/use-case-org/use-case-discvoerable|use-case-discvoerable>_",
      "type": "mrkdwn",
    },
    "type": "section",
  },
  Object {
    "type": "divider",
  },
  Object {
    "fields": Array [
      Object {
        "text": "*<https://ddw-corewebapp.dev.data.world/patrick|Patrick Star>*
*User ID*: _patrick_
*E-mail*: _patrick@star.com_
",
        "type": "mrkdwn",
      },
      Object {
        "text": "*<https://ddw-corewebapp.dev.data.world/use-case-org/use-case-discvoerable|use-case-discvoerable>*
*Resource ID*: _use-case-discvoerable_
*Owner ID*: _use-case-org_
",
        "type": "mrkdwn",
      },
    ],
    "type": "section",
  },
  Object {
    "type": "divider",
  },
]
`;

exports[`requests helper methods getAuthorizationRequestSlackBlocks should return correctly formatted blocks for authorizations created 1`] = `
Array [
  Object {
    "elements": Array [
      Object {
        "text": "üìç Use Case Org",
        "type": "mrkdwn",
      },
    ],
    "type": "context",
  },
  Object {
    "text": Object {
      "text": "_<https://ddw-corewebapp.dev.data.world/patrick|Patrick Star>_ has requested _view_ access to _<https://ddw-corewebapp.dev.data.world/use-case-org/use-case-discvoerable|use-case-discvoerable>_",
      "type": "mrkdwn",
    },
    "type": "section",
  },
  Object {
    "text": Object {
      "text": "*Request Details*
>*How long do you need access to this dataset?*:
>_6 months_
>*Any additional notes or comments?*:
>_I need access._
>*I am an existing VDAS user*:
>_Yes_
",
      "type": "mrkdwn",
    },
    "type": "section",
  },
  Object {
    "elements": Array [
      Object {
        "action_id": "authorization_request.accept",
        "style": "primary",
        "text": Object {
          "text": "Approve",
          "type": "plain_text",
        },
        "type": "button",
        "value": "{\\"requestid\\":\\"a997b017-75a5-4cbb-93b8-02f6242e9483\\",\\"agentid\\":\\"use-case-org\\",\\"datasetid\\":\\"use-case-discvoerable\\"}",
      },
      Object {
        "action_id": "authorization_request.reject",
        "style": "danger",
        "text": Object {
          "text": "Reject",
          "type": "plain_text",
        },
        "type": "button",
        "value": "{\\"requestid\\":\\"a997b017-75a5-4cbb-93b8-02f6242e9483\\",\\"agentid\\":\\"use-case-org\\",\\"datasetid\\":\\"use-case-discvoerable\\"}",
      },
    ],
    "type": "actions",
  },
  Object {
    "type": "divider",
  },
  Object {
    "fields": Array [
      Object {
        "text": "*<https://ddw-corewebapp.dev.data.world/patrick|Patrick Star>*
*User ID*: _patrick_
*E-mail*: _patrick@star.com_
",
        "type": "mrkdwn",
      },
      Object {
        "text": "*<https://ddw-corewebapp.dev.data.world/use-case-org/use-case-discvoerable|use-case-discvoerable>*
*Resource ID*: _use-case-discvoerable_
*Owner ID*: _use-case-org_
",
        "type": "mrkdwn",
      },
    ],
    "type": "section",
  },
  Object {
    "type": "divider",
  },
]
`;

exports[`requests helper methods getContributionRequestSlackBlocks should return correctly formatted blocks for contributions cancelled/rejected 1`] = `
Array [
  Object {
    "elements": Array [
      Object {
        "text": "üìç Goo Lagoon",
        "type": "mrkdwn",
      },
    ],
    "type": "context",
  },
  Object {
    "text": Object {
      "text": "_<http://localhost:3000/patrick|Patrick Star>_ has removed their suggestions to _<http://localhost:3000/goo-lagoon/catalog/item?uri=https%3A%2F%2Fgoo-lagoon.linked.data.world%2Fd%2Fddw-catalogs%2FbusinessTerm-52ab4c7d-8f88-33e3-b021-0277a37982ef|12th test>_",
      "type": "mrkdwn",
    },
    "type": "section",
  },
  Object {
    "type": "divider",
  },
  Object {
    "fields": Array [
      Object {
        "text": "*<http://localhost:3000/patrick|Patrick Star>*
*User ID*: _patrick_
*E-mail*: _patrick@star.com_
",
        "type": "mrkdwn",
      },
      Object {
        "text": "*<http://localhost:3000/goo-lagoon/catalog/item?uri=https%3A%2F%2Fgoo-lagoon.linked.data.world%2Fd%2Fddw-catalogs%2FbusinessTerm-52ab4c7d-8f88-33e3-b021-0277a37982ef|12th test>*
*Resource Type*: _Business term_
",
        "type": "mrkdwn",
      },
    ],
    "type": "section",
  },
  Object {
    "type": "divider",
  },
]
`;

exports[`requests helper methods getContributionRequestSlackBlocks should return correctly formatted blocks for contributions created 1`] = `
Array [
  Object {
    "elements": Array [
      Object {
        "text": "üìç Goo Lagoon",
        "type": "mrkdwn",
      },
    ],
    "type": "context",
  },
  Object {
    "text": Object {
      "text": "_<http://localhost:3000/patrick|Patrick Star>_ has suggested changes to _<http://localhost:3000/goo-lagoon/catalog/item?uri=https%3A%2F%2Fgoo-lagoon.linked.data.world%2Fd%2Fddw-catalogs%2FbusinessTerm-52ab4c7d-8f88-33e3-b021-0277a37982ef|12th test>_",
      "type": "mrkdwn",
    },
    "type": "section",
  },
  Object {
    "text": Object {
      "text": "*Request Details*
>*Message*:
>_These are very good changes._
",
      "type": "mrkdwn",
    },
    "type": "section",
  },
  Object {
    "text": Object {
      "text": "*<http://localhost:3000/goo-lagoon/catalog/item/approvals?uri=https%3A%2F%2Fgoo-lagoon.linked.data.world%2Fd%2Fddw-catalogs%2FbusinessTerm-52ab4c7d-8f88-33e3-b021-0277a37982ef|Manage this request ‚Üí>*",
      "type": "mrkdwn",
    },
    "type": "section",
  },
  Object {
    "type": "divider",
  },
  Object {
    "fields": Array [
      Object {
        "text": "*<http://localhost:3000/patrick|Patrick Star>*
*User ID*: _patrick_
*E-mail*: _patrick@star.com_
",
        "type": "mrkdwn",
      },
      Object {
        "text": "*<http://localhost:3000/goo-lagoon/catalog/item?uri=https%3A%2F%2Fgoo-lagoon.linked.data.world%2Fd%2Fddw-catalogs%2FbusinessTerm-52ab4c7d-8f88-33e3-b021-0277a37982ef|12th test>*
*Resource Type*: _Business term_
",
        "type": "mrkdwn",
      },
    ],
    "type": "section",
  },
  Object {
    "type": "divider",
  },
]
`;
